source/extensions/filters/network/common/redis/fault_impl.cc

```diff
@@ -0,0 +1,154 @@
+#include "extensions/filters/network/common/redis/fault_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+FaultManagerImpl::FaultImpl::FaultImpl(
+    envoy::extensions::filters::network::redis_proxy::v3::RedisProxy_RedisFault base_fault)
+    : commands_(buildCommands(base_fault)) {
+  // // Get delay
+  delay_ms_ = std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(base_fault, delay, 0));
+
+  // Get fault type
+  switch (base_fault.fault_type()) {
+  case envoy::extensions::filters::network::redis_proxy::v3::RedisProxy::RedisFault::DELAY:
+    fault_type_ = FaultType::Delay;
+    break;
+  case envoy::extensions::filters::network::redis_proxy::v3::RedisProxy::RedisFault::ERROR:
+    fault_type_ = FaultType::Error;
+    break;
+  default:
+    NOT_REACHED_GCOVR_EXCL_LINE;
+    break;
+  }
+
+  // Get the default value/runtime key
+  if (base_fault.fault_enabled().has_default_value()) {
+    if (base_fault.fault_enabled().default_value().denominator() ==
+        envoy::type::v3::FractionalPercent::HUNDRED) {
+      default_value_ = base_fault.fault_enabled().default_value().numerator();
+    } else {
+      auto denominator = ProtobufPercentHelper::fractionalPercentDenominatorToInt(
+          base_fault.fault_enabled().default_value().denominator());
+      default_value_ = (base_fault.fault_enabled().default_value().numerator() * 100) / denominator;
+    }
```

Comment:
We shouldn't normalize this to 100% which I think is what you are doing. Let's actually use the proper full runtime fractional percent as it is meant to be used. I'm pretty sure there should be utility functions elsewhere for dealing with this. If not can you add some in the same place where we do RuntimeFeatureEnabled, etc.?

Reply:
A utility function does not exist unfortunately that takes a FractionalPercent and returns an Int. I'll create it and add tests as appropriate.